parlai/crowdsourcing/projects/multisession_chat/human_eval/compile_results.py

```diff
@@ -0,0 +1,443 @@
+from parlai.crowdsourcing.tasks.model_chat.analysis.compile_results import (
+    ModelChatResultsCompiler as BaseModelChatResultsCompiler,
+)
+from parlai.crowdsourcing.utils.analysis import AbstractTurnAnnotationResultsCompiler
+
+
+class ModelChatResultsCompiler(BaseModelChatResultsCompiler):
+    """
+    Compile and save results of human+model chats.
+
+    Results will be saved on the level of specific conversations, as well as aggregated
+    up the level of each worker as a whole.
+    """
+
+    @classmethod
+    def setup_args(cls):
+        parser = super().setup_args()
+        parser.add_argument(
+            '--model-nickname', type=str, default='', help='name of the model'
+        )
+        parser.add_argument(
+            '--completed-run-stats-path',
+            type=str,
+            default='',
+            help='path of the task run stats file',
+        )
+        return parser
+
+    def __init__(self, opt: Dict[str, Any]):
+
+        AbstractTurnAnnotationResultsCompiler.__init__(self, opt)
+
+        # Input args
+        self.model_nickname = opt['model_nickname']
+        assert len(self.results_folders) > 0
+        for folder in self.results_folders:
+            assert os.path.isdir(folder), f'{folder} is not a valid folder!'
+        os.makedirs(self.output_folder, exist_ok=True)
+        self.start_date = opt['start_date']
+        self.max_convos_per_worker = opt['max_convos_per_worker']
+        self.min_word_count = opt['min_word_count']
+        self.hit_block_list = opt['hit_block_list'].split(',')
+        self.worker_block_list = opt['worker_block_list'].split(',')
+
+        # Setting up problem buckets
+        if self.use_problem_buckets:
+            self.regular_buckets = [
+                bucket
+                for bucket in self.problem_buckets
+                if bucket not in ['other', 'none_all_good']
+            ]
+            # Remove the buckets that are special cases
+
+        self.acceptability_checker = AcceptabilityChecker()
+        self.completed_run_stats_path = opt['completed_run_stats_path']
+
+    def compile_results(self) -> pd.DataFrame:
+        read_folders = []
+        date_strings = []
+        import ipdb
+
+        ipdb.set_trace()
+        for folder in self.results_folders:
+            # Load paths
+            date_strings = sorted(
+                [
+                    obj
+                    for obj in os.listdir(folder)
+                    if os.path.isdir(os.path.join(folder, obj))
+                    and re.fullmatch(r'\d\d\d\d_\d\d_\d\d', obj)
+                ]
+            )
+            if self.start_date != '':
+                date_strings = [
+                    str_ for str_ in date_strings if str_ >= self.start_date
+                ]
+            folders = [os.path.join(folder, str_) for str_ in date_strings]
+            read_folders.extend(folders)
+        print(f'Date folders: ' + ', '.join(date_strings))
+
+        # Read in each file
```

Comment:
hmm, yeah, looks like there's (understandably) a lot of repeated code here from the messy model chat analysis script. This code is in the `parlai/crowdsourcing/projects/` folder so I'm not as concerned, but hopefully someday we can modularize this code to avoid having to keep duplicating it